src/sortStudents.ts

```diff
@@ -1,16 +1,84 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const sortedStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  sortedStudents.sort((a: Student, b: Student) => {
+    let compareValue: number;
+    let avgGradeA: number;
+    let avgGradeB: number;
+
+    switch (sortBy) {
+      case SortType.Name:
+        compareValue = a.name.localeCompare(b.name);
+        break;
+
+      case SortType.Surname:
+        compareValue = a.surname.localeCompare(b.surname);
+        break;
+
+      case SortType.Age:
+        compareValue = a.age - b.age;
+        break;
+
+      case SortType.Married:
+        if (a.married === b.married) {
+          compareValue = 0;
+        } else if (a.married) {
+          compareValue = 1;
+        } else {
+          compareValue = -1;
+        }
```

Comment:
it should works the same way
```suggestion
          compareValue = Number(a.married !== b.married);
```